chore: Update to net10 pre6

jeromelaban opened the pull request:
closes https://github.com/unoplatform/uno-private/issues/1131

jeromelaban commented:
/azp run

unodevops commented:
⚠️⚠️ The build [161047](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=161047) has failed on Uno.UI - CI.

jeromelaban commented:
/azprun

jeromelaban commented:
/azp run

jeromelaban commented:
/azp run

unodevops commented:
⚠️⚠️ The build [161100](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=161100) has failed on Uno.UI - CI.

unodevops commented:
⚠️⚠️ The build [161172](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=161172) has failed on Uno.UI - CI.

unodevops commented:
⚠️⚠️ The build [161774](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=161774) has failed on Uno.UI - CI.

jeromelaban committed:
chore: Update to net10 preview 4

jeromelaban committed:
chore: Adjust template location

jeromelaban committed:
chore: Adjust template path

jeromelaban committed:
chore: Adjust global.json

jeromelaban committed:
chore: Adjust version and formatting

jeromelaban committed:
chore: Adjust base versions for mobile

jeromelaban committed:
Merge branch 'master' into dev/jela/ref-updates

unodevops commented:
⚠️⚠️ The build [165683](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=165683) has failed on Uno.UI - CI.

jeromelaban committed:
chore: Adjust for net10

jeromelaban committed:
Merge branch 'master' into dev/jela/ref-updates

jeromelaban committed:
chore: Adjust uno.check

jeromelaban committed:
chore: Disable version override

jeromelaban committed:
chore: Adjust for IDE warnings, 18.4 apple

jeromelaban committed:
chore: Adjust for net10 bootstrap

jeromelaban committed:
chore: adjust feed

jeromelaban committed:
chore: Adjust rollforward

unodevops commented:
⚠️⚠️ The build [166422](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=166422) has failed on Uno.UI - CI.

jeromelaban committed:
chore: Bump to preview 5

jeromelaban committed:
chore: merge master

unodevops commented:
⚠️⚠️ The build [169226](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=169226) has failed on Uno.UI - CI.

jeromelaban committed:
chore: Adjust formatting

unodevops commented:
⚠️⚠️ The build [169230](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=169230) has failed on Uno.UI - CI.

jeromelaban committed:
chore: Adjust formatting

jonpryor committed:
chore: Fix Tests - Android Skia > Build Samples App

Context: https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=169255&view=logs&j=afc09179-cf0c-5458-bc80-741755dcbc52&t=501b67be-3ba0-5279-cc4c-2a5532dbf9d5
Context: https://github.com/dotnet/sdk/issues/48949

The **Tests - Android Skia > Build Samples App** job is failing with:

	/agent/_work/1/s/src/Uno.UI.RemoteControl.Host/Uno.UI.RemoteControl.Host.csproj :
	error NU1102: Unable to find package Microsoft.WindowsDesktop.App.Ref with version (= 10.0.0-preview.5.25265.101)
	[/agent/_work/1/s/src/SamplesApp/SamplesApp.Skia.netcoremobile/SamplesApp.Skia.netcoremobile.csproj]

The cause of the NU1102 error is dotnet/sdk#48949, which in turn is
caused because the .NET SDK is "incoherent": most of the .NET SDK
versions are `preview.5.25277.114`, except for the
`Microsoft.WindowsDesktop.App` versions listed within
`sdk/10.0.100-preview.5.25277.114/Microsoft.NETCoreSdk.BundledVersions.props`,
which specify `10.0.0-preview.5.25265.101`.

The problem is that the [Microsoft.WindowsDesktop.App.Ref NuGet][0]
doesn't have a version `10.0.0-preview.5.25265.101`, so it cannot be
found.  NuGet *does* have version `10.0.0-preview.5.25277.114`.

Update `src/Directory.Build.targets` to update
`@(KnownFrameworkReference)` so that valid and coherent
NuGet package versions are used.  This allows
`SamplesApp.Skia.netcoremobile.csproj` to build.

[0]: https://www.nuget.org/packages/Microsoft.WindowsDesktop.App.Ref/10.0.0-preview.5.25277.114#versions-body-tab

unodevops commented:
⚠️⚠️ The build [170863](https://dev.azure.com/uno-platform///_build/results?buildId=170863) has failed on Uno.UI - docs.

jonpryor committed:
Merge remote-tracking branch 'origin/master' into dev/jela/ref-updates

jonpryor committed:
chore: remove NU1510 warning regarding System.Text.Encodings.Web

The **Tests - Android Skia > Build Android Skia Head** job fails with:

	/agent/_work/1/s/src/SamplesApp/SamplesApp.Skia.netcoremobile/SamplesApp.Skia.netcoremobile.csproj :
	error NU1510: Warning As Error: PackageReference System.Text.Encodings.Web will not be pruned. Consider removing this package from your dependencies, as it is likely unnecessary.

Removing the `@(PackageReference)` doesn't break the build, so
just remove it for now.

jonpryor committed:
chore: fix ide0055 errors

	error IDE0055: Fix formatting (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0055)

Apparently a newline is required between the namespace declaration
and the type declaration.

jonpryor committed:
chore: fix C# CS0234 compilation errors

Context: https://github.com/dotnet/sdk/issues/48949
Context: https://github.com/dotnet/android/commit/70bd636b04ebf9dba36c632cdec586452e98cb8a

Commit dotnet/android@70bd636b introduced a "source break", as it
introduced a `Microsoft.Android.Runtime` namespace, which means that
any source code that is *also* in a `Microsoft.` namespace can now
see ambiguity errors:

	src/Uno.UI/UI/Xaml/NativeApplication.cs(23,43): error CS0234: The type or namespace name 'App' does not exist in the namespace 'Microsoft.Android' (are you missing an assembly reference?)
	src/Uno.UI/UI/Xaml/NativeApplication.cs(41,89): error CS0234: The type or namespace name 'JniHandleOwnership' does not exist in the namespace 'Microsoft.Android.Runtime' (are you missing an assembly reference?)
	src/Uno.UI/UI/Xaml/NativeApplication.cs(160,55): error CS0246: The type or namespace name 'IActivityLifecycleCallbacks' could not be found (are you missing a using directive or an assembly reference?)
	…

Fix these by adding `using Android.…` namespace declarations or
`using ATypeName = …` aliases.

unodevops commented:
⚠️⚠️ The build [170879](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=170879) has failed on Uno.UI - CI.

jonpryor committed:
chore: Re-enable Uno.Resizetizer

Fixes build error APT2260, as Uno.Resizetizer is what creates the
mipmap/icon resource:

	…/uno/src/SamplesApp/SamplesApp.Skia.netcoremobile/Android/AndroidManifest.xml : error APT2260: resource mipmap/icon (aka uno.platform.samplesapp.skia:mipmap/icon) not found.
	…/uno/src/SamplesApp/SamplesApp.Skia.netcoremobile/Android/AndroidManifest.xml : error APT2260:
	…/uno/src/SamplesApp/SamplesApp.Skia.netcoremobile/Android/AndroidManifest.xml : error APT2260: This error is likely caused by an issue with the AndroidManifest.xml file or an Android manifest generation attribute in a source code file.

jonpryor committed:
chore: fix ide0055 errors

	error IDE0055: Fix formatting (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0055)

Apparently a newline is required between the namespace declaration
and the type declaration.

jonpryor committed:
chore: Fix MacCatalyst SupportedOSPlatformVersion error

The error:

	…/dotnet-sdk-10.0.100-preview.5.25277.114/packs/Microsoft.MacCatalyst.Sdk.net9.0_18.0/18.0.9617/tools/msbuild/iOS/Xamarin.Shared.targets(554,3):
	error : The SupportedOSPlatformVersion value '13.1' in the project file is lower than the minimum value '15.0'.

Bump the MacCatalyst `$(SupportedOSPlatformVersion)` value to 15.0,
which fixes the above error.

jonpryor committed:
chore: fix many more CS0234 compilation errors

Context: https://github.com/dotnet/android/commit/70bd636b04ebf9dba36c632cdec586452e98cb8a

Commit dotnet/android@70bd636b introduced a "source break", as it
introduced a `Microsoft.Android.Runtime` namespace, which means that
any source code that is *also* in a `Microsoft.` namespace can now
see ambiguity errors.

Fix these by adding `using Android.…` namespace declarations or
`using ATypeName = …` aliases.

There are three related CS0115 errors which I'm still trying to
track down:

	unoplatform/uno/src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.Android.cs(45,26): error CS0115: 'ContentPresenter.OnDraw(Canvas)': no suitable method found to override
	unoplatform/uno/src/Uno.UI/UI/Xaml/Controls/Panel/Panel.Android.cs(56,26): error CS0115: 'Panel.OnDraw(Canvas)': no suitable method found to override
	unoplatform/uno/src/Uno.UI/UI/Xaml/Controls/Border/Border.Android.cs(24,26): error CS0115: 'Border.OnDraw(Canvas)': no suitable method found to override

unodevops commented:
⚠️⚠️ The build [170988](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=170988) has failed on Uno.UI - CI.

jonpryor committed:
chore: fix CS0115 errors

Context: 41083f74fa2235d68545dde626f2c42e5e5a0ee4

Commit 41083f74 attempted to fix many CS0234 errors, but introduced
three CS0115 errors in the process:

	unoplatform/uno/src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.Android.cs(45,26): error CS0115: 'ContentPresenter.OnDraw(Canvas)': no suitable method found to override
	unoplatform/uno/src/Uno.UI/UI/Xaml/Controls/Panel/Panel.Android.cs(56,26): error CS0115: 'Panel.OnDraw(Canvas)': no suitable method found to override
	unoplatform/uno/src/Uno.UI/UI/Xaml/Controls/Border/Border.Android.cs(24,26): error CS0115: 'Border.OnDraw(Canvas)': no suitable method found to override

The cause of this is a Today I Learned: types in the same namespace
or an enclosing namespace always get priority over `using` aliases!

	using MyType = Unrelated.MyType;

	namespace Scratch.Using {
	    public class MyType {}
	}

	namespace Unrelated {
	    public class MyType {}
	}

	namespace Scratch.Using.Alias
	{
	    class App {
	        public static void Main() {
	            var t = new MyType();
	            Console.WriteLine(t.GetType().FullName);
	        }
	    }
	}

Which `MyType` does `App.Main()` use? `Scratch.Using.MyType`!

Which in turn means that 41083f74's attempt to do:

	using Canvas = Android.Graphics.Canvas;

didn't work as expected/anticipated, because it was ignored.

Which is why `OnDraw(Canvas)` couldn't override the right method,
and thus the CS0115 errors.

Review the `using-alias`es introduced in 41083f74, and if they're
an existing type such as `Canvas`, always *prefix* them so that the
resulting alias name is *unique*, e.g. `ACanvas`.

This *doesn't* explain the weirdness in `NativeWebView.Android.cs`,
which *didn't* use a `using-alias`, but would also occasionally
error out because it couldn't find the correct `WebView(Context)`
constructor that was used; I'm still not sure what's going on there,
but `using-alias` works as a fix, so…

All this does suggest a future TODO: add an API compatibility check,
to ensure that published API signatures don't change.

jonpryor committed:
chore: fix IL207x warnings in Uno.Foundation

From the **Build - Managed Binaries** stage:

	C:\a\1\s\src\Uno.Foundation\Metadata\ApiInformation.cs(104,5): error IL2075: 'this' argument does not satisfy 'DynamicallyAccessedMemberTypes.PublicProperties' in call to 'System.Type.GetProperty(String)'. The return value of method 'Windows.Foundation.Metadata.ApiInformation.GetValidType(String)' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to.
	C:\a\1\s\src\Uno.Foundation\Metadata\ApiInformation.cs(67,5): error IL2075: 'this' argument does not satisfy 'DynamicallyAccessedMemberTypes.PublicMethods' in call to 'System.Type.GetMethods()'. The return value of method 'Windows.Foundation.Metadata.ApiInformation.GetValidType(String)' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to.
	C:\a\1\s\src\Uno.Foundation\Metadata\ApiInformation.cs(71,30): error IL2070: 'this' argument does not satisfy 'DynamicallyAccessedMemberTypes.PublicEvents' in call to 'System.Type.GetEvent(String)'. The parameter 'type' of method 'Windows.Foundation.Metadata.ApiInformation.IsEventPresent(Type, String)' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to.
	C:\a\1\s\src\Uno.Foundation\Metadata\ApiInformation.cs(76,5): error IL2075: 'this' argument does not satisfy 'DynamicallyAccessedMemberTypes.PublicEvents' in call to 'System.Type.GetEvent(String)'. The return value of method 'Windows.Foundation.Metadata.ApiInformation.GetValidType(String)' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to.
	C:\a\1\s\src\Uno.Foundation\Metadata\ApiInformation.cs(91,5): error IL2075: 'this' argument does not satisfy 'DynamicallyAccessedMemberTypes.PublicProperties' in call to 'System.Type.GetProperty(String)'. The return value of method 'Windows.Foundation.Metadata.ApiInformation.GetValidType(String)' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to.

Fix these by sprinkling magic `[DynamicallyAccessedMembers(…)]`
pixie-dust in the appropriate locations.  In particular, on all
`string typeName` or `Type type` parameters, and a new
`[return: DynamicallyAccessedMembers(…)]` on `GetValidType()`.

Add a nested `AssemblyGetType()` method so we can "patch up"
the `[DynamicallyAccessedMembers]` differences between
`GetValidType()` and `Assembly.GetType()`.

jonpryor committed:
chore: Uni.UI.netcoremobile.csproj builds

Context: https://github.com/dotnet/android/commit/70bd636b04ebf9dba36c632cdec586452e98cb8a

*Ignore* the plethora of IL2* warnings; actually fixing these will be
a chore, and that's assuming they need to be fixed at this time.

Fix the various CS errors caused by the introduction of a
`Microsoft.Android` namespace within .NET for Android; see also
dotnet/android@70bd636b, and 41083f74, and…

To help address many of those errors, add a `Uno.UI/GlobalUsings.cs`
file to contain frequently used `using-alias`es, for consistency.

Fix C# nullability and `[Obsolete]` changes from .NET for iOS 10.

jonpryor committed:
chore: Fix Uno.UI.Runtime.Skia.Android.csproj build

Context: ca9231045fb10d2ecca2be75a4fa8461fdaa879c

The addition and use of `GlobalUsings.cs` in ca923104 broke the
`Uno.UI.Runtime.Skia.Android.csproj` build as it wasn't importing it:

	src/Uno.UI/UI/Xaml/NativeApplication.cs(25,35): error CS0246: The type or namespace name 'AApplication' could not be found (are you missing a using directive or an assembly reference?)

Update `Uno.UI.Runtime.Skia.Android.csproj` so that
`GlobalUsings.cs` is part of `@(Compile)`, allowing it to build.

jonpryor committed:
chore: Fix IL2073 warnings in src/Uno.UI.Toolkit

jonpryor committed:
chore: Address IL trimmer warnings in Uno.UI.XamlHost

jonpryor committed:
chore: ignore IL* trimmer warnings in Uno.UI.RemoteControl

Assuming we even need to worry about trimmer warnings in
Uno.UI.RemoteControl, the biggest pain point will be
`ClientHotReloadProcessor.UpdateApplication(Type[])`,
because [`DynamicallyAccessedMembersAttribute`][0] can only be
applied to members/parameters/etc. of type `Type` or `string`.
This *excludes* the `Typep[]` in `UpdateApplication(Type[])`.

Punt on the idea until we better know what's required.

[0]: https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.dynamicallyaccessedmembersattribute?view=net-9.0

jonpryor committed:
Merge branch 'master' into dev/jela/ref-updates

jonpryor committed:
Merge branch 'master' into dev/jela/ref-updates

jonpryor committed:
chore: Bump to Preview 6

Context: https://github.com/dotnet/macios/releases/tag/dotnet-10.0.1xx-preview6-10451
Context: https://github.com/dotnet/sdk/issues/48949

The iOS, tvOS, and maccatalyst packages for .NET 10 Preview 6 no
longer contain 18.4 packages, they now ship 18.5 packages.
This necessitates a bunch of bumps everywhere.

Update the `%(KnownFrameworkReference.DefaultRuntimeFrameworkVersion)`,
`%(LatestRuntimeFrameworkVersion)`, and `%(TargetingPackVersion)`
values to match the .NET 10 Preview 6 version, to work around
dotnet/sdk#48949:

	error NU1102: Unable to find package Microsoft.WindowsDesktop.App.Ref with version (= 10.0.0-preview.6.25315.102)
	  - Found 146 version(s) in NuGet official package source [ Nearest version: 10.0.0-preview.6.25358.103 ]

Bump `$(LangVersion)`=13.0, in order to fix:

	src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions.Collections/MemoryExtensions.cs(34,4):
	error CS9202: Feature 'ref struct interfaces' is not available in C# 12.0. Please use language version 13.0 or greater.

jonpryor committed:
chore: use build/ci/net10

Context: https://github.com/unoplatform/uno/pull/19934/
Context: https://github.com/unoplatform/uno/actions/runs/16376459008/job/46277525341?pr=19934

When reviewing the CodeQL checks for unoplatform/uno#19934,
I found that [CodeQL Advanced / Analyze (csharp)[0] was failing:

> ❌ Pin .NET Version
> 1. ➡️ Run cp build/ci/net10/global.json global.json
> 4. cp: cannot stat 'build/ci/net10/global.json': No such file or directory
> 5. Error: process completed with exit code 1

which makes a fair bit of sense, as there is no `build/ci/net10`
directory, much less `build/ci/net10/global.json`.

`dotnet-install.yml`, meanwhile, still requires `build/ci/net9`.

Rename `build/ci/net9` to `build/ci/net10`, and update
`dotnet-install.yml` to use `build/ci/net10`.  This should ensure
that everything is internally consistent, and should fix the
**CodeQL Advanced / Analyze (csharp)** job error.

[0]: https://github.com/unoplatform/uno/actions/runs/16376459008/job/46277525341?pr=19934

jonpryor committed:
chore: Bump GlobalUnoCheckVersion=1.32.0-dev.39

The **Build - Managed Binaries > Build > Building Mobile binaries**
job was [failing][0]:

	  Uno.Xaml -> C:\a\1\s\src\SourceGenerators\System.Xaml\bin\Release\netstandard2.0\Uno.Xaml.dll
	##[error].dotnet\sdk\10.0.100-preview.6.25358.103\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.TargetFrameworkInference.targets(259,5):
	Error NETSDK1140: 18.5 is not a valid TargetPlatformVersion for iOS. Valid versions include:
	18.4

This happened because the [Install .NET Workloads][1] stage was
install the preview *5* workload bits!

	SHELL: C:\a\1\s\.dotnet\dotnet.exe workload install --from-rollback-file "C:\Users\AzDevOps\AppData\Local\Temp\uno-check-65b015bd\workload.json" android ios maccatalyst tvos --source "https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet10/nuget/v3/index.json" --source "https://api.nuget.org/v3/index.json"
	…
	Installing pack Microsoft.iOS.Sdk.net10.0_18.4 version 18.4.10622-net10-p5...

@jonpryor *guesses* that the problem is that we're using
[`uno.check` 1.32.0-dev.31][2], which does not have support for
.NET 10 Preview 6.

Bump GlobalUnoCheckVersion to 1.32.0-dev.39, which *should* have
support for .NET 10 Preview 6.  Fingers crossed 🤞 it'll build now?

[0]: https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=171249&view=logs&j=76c7a511-c4f7-597f-21cf-4ad98ff8d48b&t=b7c6dcf9-e2ac-5d9e-eae5-2b124eead06d
[1]: https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=171249&view=logs&j=76c7a511-c4f7-597f-21cf-4ad98ff8d48b&t=7167d3a5-eeaa-5511-5f84-3c082f7d4916
[2]: https://www.nuget.org/packages/Uno.Check/1.32.0-dev.31

jonpryor committed:
chore: Remove `Android.` from IFrameworkElementImplementation.Android.tt

See also: ca9231045fb10d2ecca2be75a4fa8461fdaa879c, fd14b3ebd9dd62f8c44194780ec0f1dcc93545ef

The generated file `src\Uno.UI\Mixins\Android\FrameworkElementMixins.g.cs`
failed to build:

	src\Uno.UI\Mixins\Android\FrameworkElementMixins.g.cs(4168,63):
	Error CS0234: The type or namespace name 'Views' does not exist in the namespace 'Microsoft.Android' (are you missing an assembly reference?)

Update `IFrameworkElementImplementation.Android.tt` to remove
`Android.Views`.  There was already a `using Android.Views` at
toplevel, so it wasn't needed anyway.

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-19934/wasm-skia-net9/index.html

jonpryor committed:
chore: Ignore NU1510 for Microsoft.Win32.SystemEvents

Ignores:

	C:\a\1\s\src\Uno.UI.Runtime.Skia.Wpf\Uno.UI.Runtime.Skia.Wpf.csproj :
	error NU1510: Warning As Error: PackageReference Microsoft.Win32.SystemEvents will not be pruned. Consider removing this package from your dependencies, as it is likely unnecessary.

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-19934/wasm-skia-net9/index.html

unodevops commented:
⚠️⚠️ The build [171277](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=171277) has failed on Uno.UI - CI.

jonpryor committed:
chore: fix CS1705 errors

Context: eecf28ba0c477f4b40673eda0b27da5640f6e070
Context: https://github.com/dotnet/sdk/issues/48949

Building `build/filters/Uno.UI-packages-skia.slnf` would error out:

	CSC : error CS1705: Assembly 'Microsoft.Win32.SystemEvents' with identity 'Microsoft.Win32.SystemEvents, Version=10.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'
	uses 'System.Runtime, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' which has a higher version than referenced assembly 'System.Runtime'
	with identity 'System.Runtime, Version=9.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'8

This happened when building `Uno.UI.Runtime.Skia.Wpf.csproj` with
`$(TargetFramework)=net9.0`.

Where were the .NET 10 packages coming from?

The .NET 10 packages were coming from eecf28ba, which overrode e.g.
`%(KnownFrameworkReference.TargetingPackVersion)` to the "correct"
.NET 10 package version, but the way this was done *also* caused
.NET *9* builds to use the .NET 10 packages!

Oops.

Update the `Condition` on `@(KnownFrameworkReference)` so that the
fixups to workaround dotnet/sdk#48949 are only applied for
`$(TargetFramework)`s of .NET 10.

jonpryor committed:
chore: ignore NU5104 warning in Uno.UI.WebView.Skia.X11.csproj

Warning as error:

	…/dotnet-sdk-10.0.100-preview.6.25358.103-osx-x64/sdk/10.0.100-preview.6.25358.103/Sdks/NuGet.Build.Tasks.Pack/buildCrossTargeting/NuGet.Build.Tasks.Pack.targets(221,5):
	error NU5104: Warning As Error: A stable release of a package should not have a prerelease dependency.
	Either modify the version spec of dependency "WebkitGtkSharp [3.24.24.117-develop, )" or
	update the version field in the nuspec.

`Uno.UI.WebView.Skia.X11.csproj` indeed depends upon
[WebkitGtkSharp 3.24.24.117-develop][0], which indeed has
"dev" as part of the version string, which makes NuGet sad.

I wonder why this isn't a problem on master?  Feels like it should…

In the meantime, ignore warning NU5104, as it's ignored elsewhere.

[0]: https://www.nuget.org/packages/WebkitGtkSharp/3.24.24.117-develop

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-19934/wasm-skia-net9/index.html

jonpryor committed:
chore: fix Yet Another IDE0055 warning

	C:\a\1\s\src\Uno.UI\UI\Xaml\Controls\MediaPlayerElement\MediaPlayerElementExtensions.wasm.cs(4,1):
	error IDE0055: Fix formatting (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0055)

Why aren't these warnings impacting .NET 9/master?

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-19934/wasm-skia-net9/index.html

unodevops commented:
⚠️⚠️ The build [171406](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=171406) has failed on Uno.UI - CI.

jonpryor committed:
chore: what's going on with Tests - Android Skia?

The **Tests - Android Skia > Build Samples App > Install .NET Workloads**
step is failing:

	Unhandled exception: Command workload has more than one child named "--info".
	/agent/_work/_temp/eaa633b8-a851-4f41-9835-813e912ba8e5.sh: line 2: uno-check: command not found

	##[error]Bash exited with code '127'.

…but why?  It works on master!

Additionally, where is `--info` coming from?  Where is `workload`
coming from?

Add some logging to this step, so we know exactly which commands are
being executed, what `$PATH` is, and if `uno-check` is in `$PATH`.

jonpryor committed:
chore: MOAR INPUT

Context: adc71ba9d40660bbecb48e7d75a44c4987ce6bb1

Commit adc71ba9 noted confusion around "workload" and "info" in
the error message:

	Unhandled exception: Command workload has more than one child named "--info".

Now that we have `set -o xtrace`, we see that it *is* coming from
the `dotnet tool install` command:

	+ dotnet tool update --global uno.check --version 1.32.0-dev.39 --add-source https://api.nuget.org/v3/index.json
	Unhandled exception: Command workload has more than one child named "--info".

to which I say, "huh"?

Turns out, `dotnet tool` takes a verbosity flag!  Add it: let's see
what it thinks it's doing.

unodevops commented:
⚠️⚠️ The build [171470](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=171470) has failed on Uno.UI - CI.

jonpryor committed:
chore: now begins the hacking…

I'm still trying to figure out what's up with the error:

    Unhandled exception: Command workload has more than one child named "--info".

I can't repro that error locally, nor can I make
`System.CommandLine` throw it deliberately.

So as a Glorious Hack™, what if we just *replaced*
`System.CommandLine.dll` with a copy from .NET 10 beta 5?
(Not that I have any reason to believe that will be better, and the
version in .NET 10 Preview 5 *also* has the change which *could*
throw that exception…)

I'm running out of ideas.

jeromelaban committed:
ci: Move to VS17.14 compatible pool

jonpryor committed:
chore: try using $(linuxScaledPool) for Tests - Desktop Skia Linux

Context: 673cd43080228976eba85714af5558008ca31a3a

Commit 673cd430 tried a Glorious Hack™ of replacing
`System.CommandLine.dll` with the version from "beta5".

That did not work: **Tests - Desktop Skia Linux > Install .NET Workloads**
continued to fail with:

	Unhandled exception: Command workload has more than one child named "--info".

However, I noticed that **Tests - WebAssembly Native** *also* runs on
Linux, and *also* has a "Install .NET Workloads" step which executes
the same commands, and it *passes*.

Due to the `set` added in 673cd430, I saw that there were quite a few
environment differences, and it's because they use different images:
failing Desktop Skia Linux was using `$(linuxVMImage)` (`ubuntu-22.04`),
while working WebAssembly Native was using
`$(linuxScaledPool)` (`Ubuntu2204-20230918`).

Change things so that **Tests - Desktop Skia Linux** uses
`$(linuxScaledPool)`.  Let's see if this changes anything.

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-19934/wasm-skia-net9/index.html

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-19934/wasm-skia-net9/index.html

unodevops commented:
⚠️⚠️ The build [171528](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=171528) has failed on Uno.UI - CI.

jonpryor committed:
chore: cleanup "Command workload…" investigation

Partially reverts:

  * adc71ba9d40660bbecb48e7d75a44c4987ce6bb1
  * a8d1dcf45e20ecc08bd1bcdd0ab8aa546b0ba2d6
  * 673cd43080228976eba85714af5558008ca31a3a
  * b6fb106120ca18d000d35a9eb338ca26a56c6b70

Commit adc71ba9 noted a bizarre error message
from **Tests - Desktop Skia Linux > Install .NET Workloads**:

	Unhandled exception: Command workload has more than one child named "--info".

This is possibly answered on [DotNetEvolution > #sdk][0]

> **Chet Husk**:
> I'm looking for the PR but this is a bug that we fixed very recently,
> in the SDK repo itself
> https://github.com/dotnet/dotnet/pull/1385
> and that just made it into the SDK repo 2 weeks ago, so I'd expect
> preview6 to have the issue, but preview 7 to not have the issue
> it happens depending on how many times we construct the
> system.commandline parser for the dotnet CLI. difference codepaths
> would call this mechanism different ways, but because we put some
> mutability on one of the pathways it could corrupt the workload
> command parser

so this might be a Known Issue™ and will be fixed in .NET 10 Preview 7?

[0]: https://discord.com/channels/732297728826277939/1267023991428612199/1397307201919652023

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-19934/wasm-skia-net9/index.html

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-19934/docs/index.html

unodevops commented:
⚠️⚠️ The build [171545](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=171545) has failed on Uno.UI - CI.

jonpryor committed:
chore: use .NET 9 to run `dotnet tool update`?

Context: e1197f9df20f9e7f0647ae8eaaf44d969e4e9952

The current idea to fix the step
**Tests - Desktop Skia Linux > Install .NET Workloads**
is to provision *.NET 9*, and use .NET 9 to execute the
`dotnet tool update …` command.  This *should* avoid the bug
mentioned in e1197f9d…

Add a `build/ci/net9/global.json` file which requires use of the
.NET 9.0.300 SDK, and update the "Install .NET Workloads" step on
Linux to:

 1. Backup `global.json`
 2. Replace `global.json` with `build/ci/net9/global.json`
 3. Run the `dotnet tool update` command
 4. Restore the `global.json` from (1)

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-19934/docs/index.html

unodevops commented:
⚠️⚠️ The build [171620](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=171620) has failed on Uno.UI - CI.
